csrc/dynamic_transform.cpp

```diff
@@ -98,25 +92,17 @@ class DynamicTransformInitialInfoBuilder : public IterVisitor {
     auto out_tv = op->out()->as<TensorView>();
     // If there's no symbolic axis, this is a static reshape op
     if (out_tv->domain()->hasSymbolicAxis()) {
-      info_.dynamic_reshapes_.push_back(op);
+      info_.dynamic_reshapes_.push_back(out_tv);
```

Comment:
I now hold the output of these ops instead of the ops themselves, since during concretization we actually remove the original ops during `concretizeReshape`. This led to segfault in `DynamicTransform6_CUDA` when trying to implement some other changes.